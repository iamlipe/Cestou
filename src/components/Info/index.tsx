import React, {memo} from 'react';
import styled from 'styled-components/native';
import {Linking} from 'react-native';
import {OUR_SITE} from 'react-native-dotenv';

import Button from '@/components/Button';

const Info = () => {
  return (
    <StyledContainerInfo>
      <StyledText>
        Se você é uma instituição que atua no combate a fome e deseja solicitar
        seu cadastro, visite nosso site e saiba mais.
      </StyledText>
      <Button
        onPress={async () => Linking.openURL(OUR_SITE)}
        size="medium"
        buttonColor="text_only"
        textColor="primary"
        title="Ir para o site"
        noMargin
      />
    </StyledContainerInfo>
  );
};

const StyledContainerInfo = styled.View`
  width: 90%;
  align-self: center;
  justify-content: space-between;
  align-items: center;
  border: 1px solid ${({theme}) => theme.colors.PRIMARY_800};
  border-radius: 5px;
  padding: 8px;
  margin-bottom: 50px;
`;

const StyledText = styled.Text`
  font-family: ${({theme}) => theme.fonts.REGULAR_SOURCESANSPRO};
  font-size: ${({theme}) => theme.sizing.SMALLER};
  color: ${({theme}) => theme.colors.GRAY_900};
  margin-bottom: 16px;
`;

export default memo(Info);
