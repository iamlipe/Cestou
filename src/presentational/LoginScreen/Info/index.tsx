import React, {memo} from 'react';
import styled from 'styled-components/native';
import {Linking} from 'react-native';
import {OUR_SITE} from 'react-native-dotenv';
import {useTranslation} from 'react-i18next';

import Button from '@/components/Button';

const Info = () => {
  const {t} = useTranslation();

  return (
    <StyledContainerInfo>
      <StyledText>{t('text.componentInfo.title')}</StyledText>
      <Button
        onPress={async () => Linking.openURL(OUR_SITE)}
        size="small"
        buttonColor="text_only"
        textColor="primary"
        title={t('button.goToSite')}
        noMargin
      />
    </StyledContainerInfo>
  );
};

const StyledContainerInfo = styled.View`
  width: 90%;
  align-self: center;
  justify-content: space-between;
  align-items: center;
  border: 1px solid ${({theme}) => theme.colors.PRIMARY_800};
  border-radius: 5px;
  padding: 8px;
  margin-bottom: 50px;
`;

const StyledText = styled.Text`
  font-family: ${({theme}) => theme.fonts.REGULAR_SOURCESANSPRO};
  font-size: ${({theme}) => theme.sizing.SMALLER};
  color: ${({theme}) => theme.colors.GRAY_900};
  margin-bottom: 16px;
`;

export default memo(Info);
