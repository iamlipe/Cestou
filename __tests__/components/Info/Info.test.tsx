import React from 'react';
import {
  renderWithThemeProvider,
  matchSnapshotWithProvider,
} from '@/config/functions/testing';

import Info from '@/presentational/LoginScreen/Info';

describe('Info', () => {
  test('should render correctly', () => {
    renderWithThemeProvider(<Info />);
  });

  test('should matches snapshot', () => {
    matchSnapshotWithProvider(<Info />);
  });
});
