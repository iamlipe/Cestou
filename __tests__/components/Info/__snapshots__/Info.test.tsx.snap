// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info should matches snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "borderColor": "#00843F",
        "borderRadius": 5,
        "borderStyle": "solid",
        "borderWidth": 1,
        "justifyContent": "space-between",
        "marginBottom": 50,
        "paddingBottom": 8,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 8,
        "width": "90%",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#1A1A1A",
          "fontFamily": "SourceSansPro-Regular",
          "fontSize": 27,
          "marginBottom": 16,
        },
      ]
    }
  >
    Se você é uma instituição que atua no combate a fome e deseja solicitar seu cadastro, visite nosso site e saiba mais.
  </Text>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "transparent",
        "borderRadius": 4,
        "flexDirection": "row",
        "height": 38,
        "justifyContent": "center",
        "marginBottom": 0,
        "opacity": 1,
        "width": 90,
      }
    }
    testID="button"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#00843F",
            "fontFamily": "SourceSansPro-SemiBold",
            "fontSize": 27,
          },
        ]
      }
      textColor="primary"
    >
      Ir para o site
    </Text>
  </View>
</View>
`;
